app/src/main/java/com/ikarabulut/App.java

```diff
@@ -4,7 +4,7 @@
 
 public class App {
 
-    public static void main(String[] args) {
+    public static void main(String[] args) throws InvalidSymbolException {
```

Comment:
I think if something that `main` is calling could throw an exception, we'd want to make sure that we handle that exception and fail gracefully wherever it actually occurs.

Reply:
It also looks like you _are_ capturing any time this exception could occur, but I think you typed one of the methods as being able to throw this exception, even though it's captured. That's probably what caused your editor to tell you that you had to add this here.

Reply:
The real method signature for `main` methods in Java can be thought of this way:

```
public static void main(String[] args) throws RuntimeException {
}
```

However, RuntimeException and its subclasses are _unchecked_ exceptions and are implicit.

> RuntimeException is the superclass of those exceptions that can be thrown during the normal operation of the Java Virtual Machine.
> RuntimeException and its subclasses are unchecked exceptions. Unchecked exceptions do not need to be declared in a method or constructor's throws clause if they can be thrown by the execution of the method or constructor and propagate outside the method or constructor boundary.

If the compiler is telling you to add that `InvalidSymbolException` it tells me we are not catching it and handling it somewhere in the program.  Better to fix that root cause. This is not an advertisement to make everything a RuntimeException so that they can be unchecked. :)

Reply:
@riccjohn and I had a chance to get to the bottom of this. I did end up having an unchecked InvalidSymbolException but it was cause I jumped the gun while creating it. When I added the `throw` in an `if` statement the alerts came up saying to add the `throws` to the method which I did but I then finished the if conditional by adding the `try catch` which now means I don't need that heading anymore